include/miral/miral/wayland_extensions.h

```diff
@@ -92,9 +92,24 @@ class WaylandExtensions
         std::function<std::shared_ptr<void>(Context const* context)> build;
     };
 
+    /// Information that can be used to determine if to enable a conditionally enabled extension
+    /// \remark Since MirAL 3.4
+    struct Info
+    {
+        /// The application that is being given access to this extension
+        Application const& app;
+        /// If this extension is in the recommended set
+        bool recommended;
+        /// If the user has enabled this extension using the wayland-extensions or related option
+        bool manually_enabled;
+    };
+
     /// \remark Since MirAL 2.5
     using Filter = std::function<bool(Application const& app, char const* protocol)>;
 
+    /// \remark Since MirAL 3.4
+    using Condition = std::function<bool(Info const& info)>;
```

Comment:
The names `Condition` and `Info` are rather generic - nothing to suggest they are specific to protocol enablement.

Reply:
Updated